### ISSUE TITLE
Wrong IRC icon URI on hack book

### ISSUE THREAD

**Issue by Metalaka** [hoaproject/W3#135]
Originally reported by @lyrixx #134 

The URI is https://img.shields.io/badge/help-#hoaproject-ff0066.svg instead of https://img.shields.io/badge/help-hoaproject-ff0066.svg. There is an extra `#`.

Exemple on: https://hoa-project.net/En/Literature/Hack/Visitor.html

**Comment by Pierozi** [comment]
Actually is not an extra `#` It must be urlencode.
https://img.shields.io/badge/help-%23hoaproject-ff0066.svg

And the original file is correct => https://raw.githubusercontent.com/hoaproject/Visitor/master/README.md

So, I guess the issue is in the W3 Parser at https://github.com/hoaproject/W3/blob/master/Application/Resource/Literature/Hack.php

Please feel free to submit a PR if you know how to solve this.

**Comment by Metalaka** [comment]
I found where come from the issue but I don't have much free time currently, so I let someone else continue the job ;)

When parsing a link, [CommonMarkConverter](https://github.com/thephpleague/commonmark) decode [some entities](https://github.com/thephpleague/commonmark/blob/c4c8e6bf99e62d9568875d9fc3ef473fe3e18e0c/src/Util/UrlEncoder.php#L19) of the link URI (in our case `%23` is decoded to `#`).

Exemple:
```markdown
[link](https://img.shields.io/badge/help-%23hoaproject-ff0066.svg)
```
```html
<a href="https://img.shields.io/badge/help-%23hoaproject-ff0066.svg">link</a>
```

We have to open an issue on [thephpleague/commonmark](https://github.com/thephpleague/commonmark) to see if this behavior is intended or if we can help to fix the library.

**Comment by Pierozi** [comment]
Thanks @Metalaka, great research :mag:

**Comment by Hywan** [comment]
Wow… Good work guys!

**Comment by Hywan** [comment]
Actually, commonmark is doing what is expected. So maybe we should encode the `#` on our side first.

**Comment by Hywan** [comment]
I have opened an issue on thephpleague/commonmark.

**Comment by colinodell** [comment]
This has been fixed in league/commonmark version 0.15.6: https://github.com/thephpleague/commonmark/releases/tag/0.15.6

**Comment by Hywan** [comment]
Thanks @colinodell!
